src/events/member_async_handler.h

```diff
@@ -0,0 +1,27 @@
+//
+// Created by Piotr Moszkowicz on 2019-02-12.
+//
+
+#pragma once
+#include <future>
+
+#include "event.h"
+#include "function_handler.h"
+
+namespace Bald::Events {
+    template<class T, class EventType>
+    class MemberAsyncHandler : public FunctionHandler<T, EventType> {
+    public:
+        typedef void (T::*HandlerFunction)(EventType*);
+
+        constexpr MemberAsyncHandler(T* instance, HandlerFunction handler_function) noexcept : FunctionHandler<T, EventType>(instance, handler_function) {};
+    private:
+        constexpr void call(Event* e) const noexcept override {
+            auto handle = std::async([&](EventType* params) {
+                (FunctionHandler<T, EventType>::m_instance->*FunctionHandler<T, EventType>::m_handler_function)(params);
+            }, static_cast<EventType*>(e));
+            handle.get();
```

Comment:
if we immediately get a result there is no point of doing  it asynchronously at all

Reply:
Yep I thought so. We've discussed that one earlier. But it shouldn't return handle too, cuz it's void.

Reply:
I think it should return handle because handle here isn't value returned by function but std::future<void> and we may want to store that futures in some vector to make possible to call on them .get() if necessary.

Reply:
Sure, but than we can't have abstraction, we would need two vectors of handlers or more specific vector per each handler type.